### PR TITLE
fix(server): Include buckets in SourceQuantities addition

### PR DESCRIPTION
**Pull request by jjbayer**
#skip-changelog

### CODE REVIEW

File: tests/integration/test_spans.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def test_span_extraction(
     relay.send_event(project_id, event)
 
     if discard_transaction:
-        transactions_consumer.poll(timeout=2.0) is None
+        assert transactions_consumer.poll(timeout=2.0) is None
```
Comment (jjbayer) [inline tests/integration/test_spans.py:69]: This is another drive-by oopsie

File: relay-server/src/envelope.rs [hunk 2]
```diff
@@ -586,8 +586,14 @@ pub struct SourceQuantities {
...
-        self.transactions += other.transactions;
-        self.profiles += other.profiles;
+        let Self {
+            transactions,
+            profiles,
+            buckets,
+        } = self;
+        *transactions += other.transactions;
+        *profiles += other.profiles;
+        *buckets = other.buckets;
```
Comment (jjbayer) [inline relay-server/src/envelope.rs:596]: No idea how to test this to be honest. But it seems right.

**Review by Dav1dde — APPROVED** [review]
pattern matching ftw

**Review by olksdr — APPROVED** [review]
